### ISSUE TITLE
Support scan_parquet for private S3 cloud

### ISSUE THREAD

**Issue by barak1412** [pola-rs/polars#13042]
### Description

Hello,

Currently, Polars scan_parquet doesn't officially support private S3 cloud, and it leaves us with the scan_pyarrow_dataset alternative.
However, scan_pyarrow_dataset doesn't support streaming.

I checked the existing code, and altough it uses crates:object_store that supports AWS, with minor changes also private S3 cloud can work.
This code fails on closed networks:
```
scan_df = pl.scan_parquet(f's3://{bucket}/benchmark_dt/dt_partitions/**/*.parquet',
                storage_options={'access_key_id': key,
                                 'secret_access_key': secret,
                                 'endpoint_url': endpoint_url,
                },
                hive_partitioning=True)
```

This happends because when region is not supplied, Polars rust code (polars/crates/polars-io/src/cloud/options.rs:209) tries to infer it with a request to amazon AWS:

```
match region {
                Some(region) => {
                    builder = builder.with_config(AmazonS3ConfigKey::Region, region.as_str())
                },
                None => {
                    polars_warn!("'(default_)region' not set; polars will try to get it from bucket\n\nSet the region manually to silence this warning.");
                    let result = reqwest::Client::builder()
                        .build()
                        .unwrap()
                        .head(format!("https://{bucket}.s3.amazonaws.com"))
                        .send()
                        .await
                        .map_err(to_compute_err)?;
                    if let Some(region) = result.headers().get("x-amz-bucket-region") {
                        let region =
                            std::str::from_utf8(region.as_bytes()).map_err(to_compute_err)?;
                        let mut bucket_region = BUCKET_REGION.lock().unwrap();
                        bucket_region.insert(bucket.into(), region.into());
                        builder = builder.with_config(AmazonS3ConfigKey::Region, region)
                    }
                },
            };
```

Image from Proxyman:

![image](https://github.com/pola-rs/polars/assets/4324219/ef588fd7-69bb-4710-b37f-c43ebf8f2b4e)

However, if we supply some fake region in `storage_options`, e.g. , `"aws_region": "us-east-1"`, no such request is made and the code works.
I will be glad to hear your opinions.

**Comment by deanm0000** [comment]
Just to clarify, it does work but with extra nonsense. So is the request to skip the region check if the endpoint_url is supplied and in line with [this](https://stackoverflow.com/questions/77641455/how-to-limit-memory-usage-while-scanning-parquet-from-s3-and-join-using-polars/77647342#77647342) maybe set the allow_http if a endpoint_url supplied internally.

**Comment by barak1412** [comment]
@deanm0000 I know, I opened the SO question :)

Exactly, my proposal is that if endpoint_url (via environment variable or storage options) is suppied, then:
- Skip region inferene and put some default value.
- Apply allow_http automatically.

**Comment by ritchie46** [comment]
fixed by #13060
